### DISCUSSION TITLE
[V5] Command not waiting for Events to finish

### DISCUSSION THREAD

**Discussion by danpospisil** [adonisjs/core#2111] [Help] [answered]
I am trying to move my App to v5 and I came across one difference from v4.1 - I have one command that is using Events. In v4.1, the command is waiting for all events to finish but in v5, when command finishes it terminates even though that events did not finish yet. So what is the new way how to wait to all events to finish before terminating? 

Code I am using to demonstrate:
**Commands/Test.ts:**
```
import { BaseCommand } from '@adonisjs/ace'
import Event from '@ioc:Adonis/Core/Event'

export default class Test extends BaseCommand {

public static settings = {
	loadApp: true
}

  public static commandName = 'test'

  public static description = ''

  public async run () {
	Event.emit('test:test', {})
	console.log('Command done')
  }
}
```
**Listeners/TestEvent.ts:**
```
export default class TestEvent {
    public async test() {
        console.log("Event start")
        await new Promise(resolve => setTimeout(resolve, 2000))
        console.log("Event done")
    }
}
```
**Command output:**
```
Command done
Event start
```
If I add `await new Promise(resolve => setTimeout(resolve, 10000))` before `console.log('Command done')` the event has time to finish - but that's not very reliable way.

**Comment by dievardump** [comment, marked as answer]
From the event tests, Event.emit seems to return a Promise. https://github.com/adonisjs/events/blob/develop/test/event.spec.ts

Have you tried: 

`await Event.emit('test:test', {})` ?

**Reply by danpospisil** [reply]
Yeah that works, but I am using events for firing multiple I/O intensive tasks for every user (sync tasks) so awaiting for every single one will make the command run much longer compared to v4.1

**Reply by dievardump** [reply]
Do an `await Promise.all(events)` on all events then, so they can work at the same time

**Reply by danpospisil** [reply]
The events are now fired on the fly, but I guess I can rewrite the code so I can await them all at once. I hope docs on the commands will be written soon. Thanks though :)

**Reply by thetutlage** [reply]
You can mark your command to stayAlive. It means AdonisJS will not kill the process implicitly. 

```ts
public static settings = {
  loadApp: true,
  stayAlive: true
}
```
